R/predl_chrome_for_testing.R

```diff
@@ -0,0 +1,59 @@
+#' Pre-download function for Chrome-for-testing site
+#'
+#' @param url URL of the JSON to use for files
+#' @param platform One of `c("linux64", "mac-arm64", "mac-x64",
+#'                           "win32", "win64")
+#' @param history Integer number of recent entries
+#' @param appname Name of the app, typically `"chromedriver"`
+#' @param platformregex A filter for platforms. Defaults to the platform names.
+#'
+#' @return A named list of dataframes.  The name indicates the platform.  The dataframe should contain the version, url, and file to
+#' be processed.  Used as input for `binman::download_files()` or an equivalent.
+#' @importFrom binman assign_directory
+#' @export
+predl_chrome_for_testing <- function(url, platform, history,
+                                     appname,
+                                     platformregex = platform) {
+  assert_that(is_URL_file(url))
+  assert_that(is_character(platform))
+  platform <- c(mac64_m1 = "mac-arm64",
+                mac64    = "mac-x64",
+                linux32  = "linux32",
+                linux64  = "linux64",
+                win32    = "win64")[platform]
+  assert_that(is_integer(history))
+  assert_that(is_string(appname))
+  assert_that(is_character(platformregex))
+  ver_data <- jsonlite::fromJSON(url)[[2]]
+  ver_data <- Filter(function(x) !is.null(x$downloads[[appname]]),
+         ver_data)
+  ver_data <- ver_data[order(as.numeric(names(ver_data)))]
+  unwrap <- function(entry) {
+    version <- entry$version
+    downloads <- entry$downloads[[appname]]
+    if (!is.null(downloads)) {
+      platform <- downloads$platform
+      url <- downloads$url
+      file <- basename(url)
+      data.frame(version, platform, url, file)
+    }
+  }
+  extracted <- do.call(rbind, lapply(ver_data, unwrap))
+  app_links <- tapply(extracted, extracted$platform, identity)
+  app_links <- app_links[platform]
+  names(app_links) <- c("mac-arm64" = "mac64_m1",
+                        "mac-x64" = "mac64",
+                        linux32  = "linux32",
+                        linux64  = "linux64",
+                        win64    = "win32")[names(app_links)]
+  assign_directory(app_links, appname)
+}
+
+unziptar_dlfiles <- function(...) {
```

Comment:
You need to document and export this function, since you've used it in `chromedriver.yml` as `wdman::unziptar_dlfiles`. For example:

```r
#' Unzip/untar files
#'
#' Unzip or untar downloaded files
#'
#' @param ... Passed into [binman::unziptar_dlfiles].
#'
#' @return The same as [binman::unziptar_dlfiles].
#'
#' @export
unziptar_dlfiles <- function(...) {
```